Lib/importlib/_bootstrap_external.py

```diff
@@ -209,7 +209,9 @@ def _write_atomic(path, data, mode=0o666):
         # We first write data to a temporary file, and then use os.replace() to
         # perform an atomic rename.
         with _io.FileIO(fd, 'wb') as file:
-            file.write(data)
+            bytes_written = file.write(data)
+        if bytes_written != len(data):
+            raise OSError("os.write didn't write the full pyc file")
```

Comment:
```suggestion
            raise OSError("os.write() didn't write the full .pyc file")
```

Reply:
I'm trying to decide if `OSError` is the best exception or `ImportError`? While it does seem like the OS is the reason the write didn't fully succeed, it's import deciding that writing less isn't acceptable.

Reply:
`OSError` is what surrounding code expects / the except after this catches to try and remove the partially-written file.

Reply:
> `OSError` is what surrounding code expects / the except after this catches to try and remove the partially-written file.

exactly. I've added a comment to explain why I'm using an `OSError`. The approach I'm taking is the same as how we handle it if we don't have write permissions in the directory or some other OSError happens on the pyc write. We don't write the pyc file and clean up as much as possible. but otherwise we don't raise a user-visible exception, the `OSError` is handled by the caller of `_write_atomic`.